File: lab/rust/add.rs
struct Point {
	x: i32,
	y: i32,
}

fn add(a: i32, b: i32) -> i32
{
	a + b
}

fn main() {
	let my_point : Point;
	// my_point.x = 3;
	// my_point.y = 7;
	
	// my_point = {2, 7}; // wrong syntax
	// my_point = Point {2, 7}; // wrong syntax - expected identifier
	my_point = Point {x: 2, y: 7};

	println!("Point x: {}: Point y: {}!", my_point.x, my_point.y);

	let sum = add(my_point.x, my_point.y);
	println!("Sum: {}", sum);

}

File: lab/explorer/cpp/add.cpp
#include <iostream>
#include <string>

struct Point
{
	int x;
	int y;
};


int add(int a, int b)
{
	return a + b;
}


int main()
{

	std::string name;
	Point my_point;
	// my_point.x = 3;
	// my_point.y = 7;

	int sum;
	sum = add(my_point.x, my_point.y);

	std::cout << "Point x: "  << my_point.x << "; Point y: " << my_point.y << "!" << std::endl;
	std::cout << "Sum: " << sum << std::endl;

	return 0;
}
